src/hotspot/share/classfile/classFileParser.cpp

```diff
@@ -3911,18 +3919,6 @@ void ClassFileParser::parse_classfile_attributes(const ClassFileStream* const cf
               permitted_subclasses_attribute_length = attribute_length;
             }
             cfs->skip_u1(attribute_length, CHECK);
-
-          } else if (_major_version >= JAVA_16_VERSION) {
-            if (tag == vmSymbols::tag_record()) {
-              if (parsed_record_attribute) {
-                classfile_parse_error("Multiple Record attributes in class file %s", THREAD);
-                return;
-              }
-              parsed_record_attribute = true;
-              record_attribute_start = cfs->current();
-              record_attribute_length = attribute_length;
-              }
-            cfs->skip_u1(attribute_length, CHECK);
```

Comment:
Minor nit, since records, permitted subclasses and an unknown attribute type all do a "cfs->skip_u1(attribute_length, CHECK);" as the last action, you could pull that out to the end of the else if (_major_version >= JAVA_16_VERSION) statement.

Reply:
Thanks for looking at this.  Please see latest commit that contains the changes you suggested.